protocol/auditor.go

```diff
@@ -0,0 +1,55 @@
+// This module implements a generic CONIKS auditor, i.e. the
+// functionality that clients and auditors need to verify
+// a server's STR history.
```

Comment:
Yes, I intended to have #170 merged first, and this will be rebase on top of that.